Wrappers/Python/cil/processors/MaskGenerator.py

```diff
@@ -0,0 +1,334 @@
+                mask[tmp > self.threshold_factor * c * numpy.median(tmp)] = 0
+            
+        elif self.mode == 'movmean':
+
+            # if movmean along specific axis     
+            if self.axis is not None:
+                
+                axis = data.get_dimension_axis(self.axis)
+                ndim = len(data.dimension_labels)
+                
+                kernel = [1] * ndim
+                kernel[axis] = self.window
+                kernel = tuple(kernel)
+
+                mean_array = ndimage.generic_filter(data.as_array(), numpy.mean, size=kernel, mode='reflect')
+                std_array = ndimage.generic_filter(data.as_array(), numpy.std, size=kernel, mode='reflect')
+
+                mask[numpy.abs(data.as_array() - mean_array) > self.threshold_factor * std_array] = 0
+            
+            # if global movmean
+            else:
+                
+                ndim = len(data.dimension_labels)
+
+                mean_array = ndimage.generic_filter(data.as_array(), numpy.mean, size=(self.window,)*ndim, mode='reflect')
+                std_array = ndimage.generic_filter(data.as_array(), numpy.std, size=(self.window,)*ndim, mode='reflect')
+                
+                mask[numpy.abs(data.as_array() - mean_array) > self.threshold_factor * std_array] = 0          
+            
+        elif self.mode == 'movmedian':
+
+            c = -1 / (numpy.sqrt(2) * special.erfcinv(3 / 2))
+            
+            # if movmedian along specific axis
+            if self.axis is not None:
+                
+                axis = data.get_dimension_axis(self.axis)
+                ndim = len(data.dimension_labels)
+                
+                # construct filter kernel
+                kernel_shape = []
+                for i in range(ndim):
+                    if i == axis:
+                        kernel_shape.append(self.window)
+                    else:
+                        kernel_shape.append(1)
+                
+                kernel_shape = tuple(kernel_shape)
+                
+                median_array = ndimage.median_filter(data.as_array(), footprint=kernel_shape, mode='reflect')
+                
+                tmp = abs(data.as_array() - median_array)
+                mask[tmp > self.threshold_factor * c * ndimage.median_filter(tmp, footprint=kernel_shape, mode='reflect')] = 0
+            
+            # if global movmedian
+            else:
+                
+                # construct filter kernel
+                ndim = len(data.dimension_labels)
+                kernel_shape = tuple([self.window]*ndim)
+                median_array = ndimage.median_filter(data.as_array(), size=kernel_shape, mode='reflect')
+                
+                tmp = abs(data.as_array() - median_array)
+                mask[tmp > self.threshold_factor * c * ndimage.median_filter(tmp, size=kernel_shape, mode='reflect')] = 0
+              
```

Comment:
You should finish if/if else statements with an else to catch anything else. Here a `raise ValueError('Mode not recognised') `type of message.